### DISCUSSION TITLE
ESP8266 vs ESP32: Different serial communication behavior with MielHVAC device

### DISCUSSION THREAD

**Discussion by host-felice** [arendst/Tasmota#19238] [Support request]
Hello, I need some help with the serial communication between an ESP8266 and an ESP32 board with a MielHVAC device (Mitsubishi AC). I have the same connections for both boards (I have made a custom pcb with a level shifter and I can use an esp32 or an esp8266), but I get different results.

For the ESP8266, I use the following configuration: tx -> mielhvac tx // rx -> mielhvacs rx
<img width="263" alt="2023-08-01" src="https://github.com/arendst/Tasmota/assets/36647269/52bd33f6-dfa3-4a2f-9d0f-c5ba8c5644c1">

This works fine and I get “SNS: Hardware Serial” in the console:
<img width="637" alt="2023-08-01 (1)" src="https://github.com/arendst/Tasmota/assets/36647269/24e12548-5de1-4dec-908e-10920b861b47">

For the ESP32, I use the same configuration, but it does not work. The ESP32 does not detect the MielHVAC device. I have to either swap the wires or swap the software configuration to make it work:

tx -> mielhvac rx // rx -> mielhvac tx
<img width="268" alt="2023-08-01 (2)" src="https://github.com/arendst/Tasmota/assets/36647269/4e09add2-0c74-4786-87c7-9bcd050b6941">

However, I do not get any confirmation of the hardware serial being active.

Both esp8266 and esp32 run latest development 13.0.0.3 to fix the switch error self compiled with tasmo compiler.

I wonder why this is happening and what is the correct way to connect and configure the ESP32 for hardware serial communication with the MielHVAC device.

Can anyone please help me understand this issue?

Thank you very much.

**Comment by host-felice** [comment]
My mistake, ESP32 is TTL so the connection must be tx-tx and rx-rx

**Reply by barbudor** [reply]
On ESP32 is is better to use other gpios (actually any other gpios) in order to leverage the additional UART of the ESP32 without cloaking the console
On esp8266 you have no choice

**Reply by host-felice** [reply]
thank you @barbudor I switched to gpio16 and gpio17 but I still get some strange messages. I made another post here: https://github.com/arendst/Tasmota/discussions/19239

Any idea?
